### ISSUE TITLE
Windows snapping layouts

### ISSUE THREAD

**Issue by RiccardoManzan** [microsoft/wslg#727]
**Is your feature request related to a problem? Please describe.**
I have a WSL2 + WSLg on a windows 11 machine
I would like to use window snapping layouts with WSL-installed applications

**Describe the solution you'd like**
I would like to use windows snapping layouts when using WSL-installed applications

**Describe alternatives you've considered**
none

**Additional context**
none

**Comment by sergju** [comment]
Do you mean like some other x11 servers do, x410 for example?
![изображение](https://user-images.githubusercontent.com/25798315/167085779-a78bcdcf-e987-4401-a530-eb78e149375d.png)
This is for firefox running from wsl2.

**Comment by RiccardoManzan** [comment]
Exactly!
This is not available for me when using intellij.

**Comment by sergju** [comment]
Yes, that is why I am using external x server x410 and waiting the same functional from WSLg.

**Comment by elsaco** [comment]
@RiccardoManzan try this: https://docs.microsoft.com/en-us/windows/powertoys/fancyzones

**Comment by hideyukn88** [comment]
This requires local move support in remote desktop protocol, and I'm working on this. However, given this also requires some changes in Window's side RDP client software, it requires some time to be completed, thanks!

**Comment by mrconnerton** [comment]
@hideyukn88 does powertoys fancyzones fall into the second category or would it require it's own separate support?

**Comment by morphykuffour** [comment]
> Yes, that is why I am using external x server x410 and waiting the same functional from WSLg.

@sergju I have X410 but I don't have this feature. what settings do I have to change to get that same functionality?

**Comment by mrconnerton** [comment]
I have in my `~/bashrc`
```sh
export DISPLAY=$(awk '/nameserver / {print $2; exit}' /etc/resolv.conf 2>/dev/null):0.0
export LIBGL_ALWAYS_INDIRECT=1
```

**Comment by mangkoran** [comment]
@mrconnerton could you explain what your [code](https://github.com/microsoft/wslg/issues/727#issuecomment-1151374555) do?

**Comment by mrconnerton-encorehc** [comment]
@mangkoran it grabs the ip from /etc/resolv.conf, which is the wsl ethernet adapter in windows, and uses it to set the X11 display env variable to ip_address:0.0. Honestly I can't remember why I have LIBGL_ALWAYS_INDIRECT set, but I remember having issues without it. So might be optional.

**Comment by crotoc** [comment]
Looking forward to this feature. Has waited for this feature for more than a year.

**Comment by crotoc** [comment]
> Yes, that is why I am using external x server x410 and waiting thI e same functional from WSLg.

I can not do this using x410. Can I know what extra setting have you done?

**Comment by eugenov** [comment]
> 

You just need to create `%userprofile%\.wslconfig` file with
```
[wsl2]
guiApplications=false
```
inside and restart wsl. This will turn off wslg

**Comment by crotoc** [comment]
> > 
> 
> You just need to create `%userprofile%\.wslconfig` file with
> 
> ```
> [wsl2]
> guiApplications=false
> ```
> 
> inside and restart wsl. This will turn off wslg. Thus you can use normal x11 server, such as x410 that support snapping.

Thanks very much! Suceed! : )

**Comment by sergju** [comment]
I don't disable wslg to use x411. Just change display variable, but need to change gdk backend to x11

export MY_EXT_IP=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2; exit;}')
export DISPLAY=$MY_EXT_IP:0.0
export GDK_BACKEND=x11

**Comment by crotoc** [comment]
> I don't disable wslg to use x410. Just change display variable, but need to change gdk backend to x11. I think without changing backend wsl continues using wslg
> 
> export MY_EXT_IP=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2; exit;}') export DISPLAY=$MY_EXT_IP:0.0 export GDK_BACKEND=x11
> 
> Sometimes comment this to check wslg behaviour

That's great to only revise files in wsl2. Thanks!

**Comment by RiccardoManzan** [comment]
I tried a VcxSrv and yeah, it's working but there are some other issues i'm not gonna deal with.

As already said, I think this is something very related to #22, and in that issue, much older than this one, there hasn't been any update lately.
Is there any planning about this functionality?

**Comment by erncabgppml057** [comment]
is anything updates?

**Comment by benhillis** [comment]
-

**Comment by Jackenmen** [comment]
It's marked as a duplicate, presumably of #22.
Edit: the comment I replied to has been removed.
